doc/docs/modules/ROOT/pages/quickstart_writer.adoc

```diff
@@ -215,4 +215,88 @@ UNwIND $events AS event
 CREATE (n:Person{fullName: event.name + event.surname})
 ----
 
-Where `event` represents each Dataset row.
\ No newline at end of file
+Where `event` represents each Dataset row.
+
+==== Schema Optimization Operations
+
+The spark connector supports schema optimization operations via:
+
+* index
+* constraints
+* set of schema queries
+
+that will be executed *before* the import process will start in order to speed-up the import itself.
+
+You can set the optimization via `schema.optimization.type` option that takes three values:
+
+* `INDEX`: it creates only indexes on provided nodes
+* `CONSTRAINTS`: it creates only indexes on provided nodes
+* `QUERY`: it perform a series of schema queries separated by `;`
+
+and it works only when you're merging nodes.
+
+===== Index Creation
+
+Following an example of how to create indexes while you're creating nodes
+
+----
+ds.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Overwrite)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", ":Person:Customer")
+      .option("node.keys", "surname")
+      .option("schema.optimization.type", "INDEX")
+      .save()
+----
+
+This will create, before the import starts, the following schema query:
+
+----
+CREATE INDEX ON :Person(surname)
+----
+
+*So please into consideration that the first label is used for the index creation*
+
+
+===== Constraint Creation
+
+Following an example of how to create indexes while you're creating nodes
+
+----
+ds.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Overwrite)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", ":Person:Customer")
+      .option("node.keys", "surname")
+      .option("schema.optimization.type", "CONSTRAINT")
+      .save()
+----
+
+This will create, before the import starts, the following schema query:
+
+----
+CREATE CONSTRAINT ON (p:Person) ASSERT (p.surname) IS UNIQUE
+----
+
+*So please into consideration that the first label is used for the index creation*
+
+===== Index/Constraint Creation via schema query
+
+Following an example of how to create indexes while you're creating nodes
+
+----
+ds.write
```

Comment:
this is a bit different than the ticket asks for; here you do a bunch of queries ahead of doing a write, but it's all part of the write.   In this mode, it would be very hard to handle failure of idempotent index creation for example.

Why tie it to the write or read, rather than making the querys another write/read?

Reply:
Actually the will be no failure in case the index is already present we log the creation operation as `KEPT`